File: backend/app/schemas/log.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class PublishLogEntry(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    attempted_at: Optional[datetime] = None
    page_name: str = ""
    content_preview: str = ""
    result: str
    fb_post_id: Optional[str] = None
    error_code: Optional[int] = None
    error_message: Optional[str] = None
    retry_count: int = 0


class PublishLogResponse(BaseModel):
    entries: list[PublishLogEntry]
    total: int

File: frontend/src/types/log.ts
export interface PublishLogEntry {
  attempted_at: string
  page_name: string
  content_preview: string
  result: string
  fb_post_id?: string
  error_code?: number
  error_message?: string
  retry_count: number
}

export interface LogResponse {
  entries: PublishLogEntry[]
  total: number
}
